### ISSUE TITLE
04-22 9% tps performance regression on batch_insert benchmark

### ISSUE THREAD

**Issue by mahjonp** [tikv/tikv#10077]
### What version of TiKV are you using?

e769142a9fec4dcd4a5832d791c0464c639f0906

### Steps to reproduce

https://github.com/pingcap/productivity-tools/tree/master/benchmark/batch_insert

```
Version:
tidb: 8d1c2f8cf257612ffa7f2280a4bf384f5f49a814(master) 2021-04-21 15:21:29
tikv: e769142a9fec4dcd4a5832d791c0464c639f0906(master) 2021-04-22 14:52:58
pd: 85d588391bc2fc4265220c50be41f77e7f5b347f(master) 2021-04-22 05:46:45

Compare:
tidb: https://github.com/pingcap/tidb/compare/8d1c2f8...8d1c2f8
tikv: https://github.com/tikv/tikv/compare/30aaa35...e769142
pd: https://github.com/pingcap/pd/compare/85d5883...85d5883

tidb_max_cpu: 16.58, delta: -10.71%
tikv_max_cpu: 9.51, delta: -5.63%
tidb_max_memory: 1737.11 MiB, delta: -0.40%
tikv_max_memory: 23250.98 MiB, delta: 0.78%

tps: 532.45 ± 0.1559% (std=0.83) delta: -9.34%
qps: 532.45 ± 0.1559% (std=0.83) delta: -9.34%
lat_avg: 480.64 ± 0.1519% (std=0.73) delta: 10.27%
lat_99th: 746.44 ± 1.8005% (std=13.44) delta: 2.65%
```

**Comment by 5kbpers** [comment]
Trying to fix it in #10139

**Comment by 5kbpers** [comment]
Fixed by https://github.com/tikv/tikv/pull/10245
